docs/architecture/maui/accessing-remote-data.md

```diff
@@ -1,9 +1,380 @@
+
+When the `CatalogView` is navigated to, the `OnInitialize` method in the CatalogViewModel class is called. This method retrieves catalog data from the catalog microservice, as demonstrated in the following code example:
+
+```csharp
+public override async Task InitializeAsync(object navigationData)
+{
+    Products = await _productsService.GetCatalogAsync();
+} 
+```
+
+This method calls the `GetCatalogAsync` method of the `CatalogService` instance that was injected into the `CatalogViewModel` by the dependency injection container. The following code example shows the `GetCatalogAsync` method:
+
+```csharp
+public async Task<ObservableCollection<CatalogItem>> GetCatalogAsync()
+{
+    UriBuilder builder = new UriBuilder(GlobalSetting.Instance.CatalogEndpoint);
+    builder.Path = "api/v1/catalog/items";
+    string uri = builder.ToString();
+
+    CatalogRoot catalog = await _requestProvider.GetAsync<CatalogRoot>(uri);
+
+    return catalog?.Data;          
+} 
+```
+
+This method builds the URI that identifies the resource the request will be sent to, and uses the `RequestProvider` class to invoke the GET HTTP method on the resource, before returning the results to the `CatalogViewModel`. The `RequestProvider` class contains functionality that submits a request in the form of a URI that identifies a resource, an HTTP method that indicates the operation to be performed on that resource, and a body that contains any data required to perform the operation. For information about how the `RequestProvider` class is injected into the `CatalogService` class, see [Dependency Injection](dependency-injection.md).
+
+The following code example shows the `GetAsync` method in the `RequestProvider` class:
+
+```csharp
+public async Task<TResult> GetAsync<TResult>(string uri, string token = "")
+{
+    HttpClient httpClient = CreateHttpClient(token);
+    HttpResponseMessage response = await httpClient.GetAsync(uri);
+
+    await HandleResponse(response);
+    string serialized = await response.Content.ReadAsStringAsync();
+
+    TResult result = await Task.Run(() => 
+        JsonConvert.DeserializeObject<TResult>(serialized, _serializerSettings));
+
+    return result;
+}
+```
+
+This method calls the `CreateHttpClient` method, which returns an instance of the `HttpClient` class with the appropriate headers set. It then submits an asynchronous `GET` request to the resource identified by the URI, with the response being stored in the `HttpResponseMessage` instance. The `HandleResponse` method is then invoked, which throws an exception if the response doesn't include a success HTTP status code. Then the response is read as a string, converted from JSON to a `CatalogRoot` object, and returned to the `CatalogService`.
+
+The `CreateHttpClient` method is shown in the following code example:
+
+```csharp
+private HttpClient _httpClient;
+
+private HttpClient CreateHttpClient(string token = "")
```

Comment:
Might want to consider using `Lazy<HttpClient>`. Also, I believe there is a bug in this code, in that multiple calls could mistakenly reuse an old auth token in a request. For example, request A passes a token, and then request B doesn't require one, but uses the cached token header from a previous. You might want to clear that out.